fix(mechanics): Fix bug where over-capacity fleets don't unload all their cargo

Anarchist2 opened the pull request:
## Summary
If you try to run `PoolCargo()` while you have more cargo than you have cargo space, then not all your cargo ends up being unloaded from your ships. This leads to some bugs, as seen by following these instructions:
1. Start a new save and buy a Star Barge.
2. Get more than 20 tons of cargo by buying commodities/accepting jobs.
3. Sell your Star Barge and buy a Shuttle.
4. Attempt to take off and back out of the warning dialog saying that you don't have enough cargo capacity.
5. Notice that the amount of free cargo space you have has suddenly increased.
6. Attempt to take off again, and see that the take-off warning lists you as needing to get rid of more cargo than the trade panel says you do.

This PR fixes this bug by temporarily uncapping the player's cargo and bunk capacity while pooling their cargo after a failed take-off attempt.

## Testing Done
Followed the numbered instructions above with this PR enabled and encountered no bugs.

## Performance Impact
When selecting "no" on the take-off warning dialog, the game now runs `SetSize()`, `SetBunks()` and `UpdateCargoCapacities()`.

tibetiroka commented:
The displayed number is still wrong while the popup is active.

TomGoodIdea commented:
The current solution looks quite hacky. Maybe adding a flag to the transfer method forcing the operation even with not enough space would be more reliable?

Anarchist2 commented:
`SetSize(-1)` is what the game already does to force outfits into a full cargo hold.
https://github.com/endless-sky/endless-sky/blob/ec9746cea73e985320510aeb16a6202335ee4f94/source/GameAction.cpp#L100-L104

Moved the cargo size adjustments to be part of `PoolCargo()`, and moved `PoolCargo()` to before the prompt shows up (and added `DistributeCargo()` after choosing to take off) so that the free space number on the trade panel doesn't change while the popup is active. `PoolCargo()` doesn't work outside of a planet anyway, so you shouldn't be restricted by the size of your hold.

warp-core commented on source/PlanetPanel.cpp:
```diff
@@ -452,9 +455,9 @@ void PlanetPanel::CheckWarningsAndTakeOff()
-		TakeOff(false);
+		TakeOff(true);
 	else
-		player.PoolCargo();
+		return;
```
We don't need the "else" branch at all if there's nothing to do there.
